pages/extensions/amp_dependencies/__init__.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+import uuid
+from sets import Set
+
+from grow import extensions
+from grow.documents import document, static_document
+from grow.extensions import hooks
+
+class AmpDependencies(object):
+    def __init__(self, doc):
+        self._doc = doc
+        self._pod = doc.pod
+        # Used to determine where to print all dependencies
+        self._placeholder = '{}'.format(uuid.uuid4())
+        # Stores registered paths
+        self._dependencies = []
+
+    def __repr__(self):
+        return '<AmpDependencies({})>'.format(self._placeholder)
+
+    def add(self, name, version='0.1'):
+        src = 'https://cdn.ampproject.org/v0/{name}-{version}.js'.format(name=name, version=version)
+        type = 'element' if name != 'amp-mustache' else 'template'
```

Comment:
When we are changing things to an explicit mode I think it would be better to explicitly define the type.
I would prefer to have different methods for templates and elements.
We already have that for [example] tags (different attributes 'imports' and 'template')
New template implementations would be supported out of the box then.

Reply:
Added the optional kwarg `type='element'`.